### DISCUSSION TITLE
Set up and manually trigger of a GitLab pipeline

### DISCUSSION THREAD

**Discussion by VanessaScherma** [INTO-CPS-Association/DTaaS#763] [General]
The objective is the set up and manually trigger of a GitLab pipeline.

This file, structured in a YAML format, will include different jobs, including 'client'. Each job within the YAML file will encompass specific steps, corresponding to distinct tasks.

- The first step will be the creation of the structure in the Gitlab project. This include generating the yaml file to perform some steps and saving it in the Gitlab repository. Any subsequent push to the repository will automatically trigger the execution of this yaml file.

- The second step is to create the API call, which will be sent from the website to manually trigger the pipeline.

This pipeline will be integrated as a library function in the client. Given that files in Library are stored in a Git repository, the final goal is to populate the repository of the respective user with digital twins, and send the API call to trigger the execution of that code.

**Comment by prasadtalasila** [comment]
### TODOs for Prasad

* add Vanessa as developer to [user1](https://gitlab.com/dtaas/user1) repository.
* ~~Push **mass-spring-damper** DT example into `user1/Digital_twins/mass-spring-damper`~~ (done)

### General comments:

1. [git beaker](https://github.com/jdalrymple/gitbeaker) to be used as gitlab client
2. All git beaker code to be put into `client/utils/gitlab`

**Comment by prasadtalasila** [comment]
> ### TODOs for Prasad
> 
>     * add Vanessa as developer to [user1](https://gitlab.com/dtaas/user1) repository.
access granted.
